src/components/ProductCard.tsx

```
@@ -26,60 +30,85 @@ interface Product {
+    if (!user) {
+      toast({
+        title: "Login required",
+        description: "You must be logged in to upvote products.",
+        variant: "destructive"
+      });
+      return; // Exit if user is not logged in
+    }
 
-  const handleUpvote = () => {
-    if (currentUser.dailyLimits.upvoting.used >= currentUser.dailyLimits.upvoting.max) {
+    if (user && user.dailyLimits && user.dailyLimits.upvoting.used >= user.dailyLimits.upvoting.max) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)